src/ipc/ipc3/handler.c

```diff
@@ -661,7 +661,7 @@ static int ipc_pm_context_save(uint32_t header)
 	/* TODO: mask ALL platform interrupts except DMA */
 
 	/* mask all DSP interrupts */
-	arch_interrupt_disable_mask(0xffffffff);
+	arch_irq_lock();
```

Comment:
This looks a bit weird as unmatched lock pair, but I guess this is ok given we are going to D3. OS size code to put system to D3 might probably be a more appropriate place to mask the interrupts (versus the IPC app).